### DISCUSSION TITLE
Questions concerning discrete adjoint optimization of turbulence models with multiple parameters

### DISCUSSION THREAD

**Discussion by lpz456** [mdolab/dafoam#625] [General]
Hello, Professor He!

I've been considering some modification to turbulence models based on DAFoam recently. Regarding discrete adjoint optimization, I have the following question:

Firstly, suppose we only multiply the correction factor α on the production term Pk of the turbulent kinetic energy k equation. Based on the time-averaged velocity in LES, performing discrete adjoint optimization on α, we can obtain an α1 field that minimizes the error between the predicted velocity and LES-averaged results;

Similarly, suppose we only multiply the correction factor β on the dissipation term Dk of the turbulent kinetic energy k equation. Based on the time-averaged velocity in LES, performing discrete adjoint optimization on β, we can obtain a β1 field that also minimizes the error between the predicted velocity and LES-averaged results;

The question is, if we simultaneously introduce correction factors α and β in the turbulent kinetic energy equation, what would the α field and β field generated by adjoint optimization look like? Would it be a multi-solution problem? For example, (α=α1, β=1), or (α=1, β=β1), or would α and β converge to fields (α=α2, β=β2) respectively?

Looking forward to your insights on this matter.

**Comment by friedenhe** [comment]
If you optimize two factors simultaneously, they will converge to their respective optimal fields (α=α2, β=β2). This capability is already there. In the last version, we added two augmented terms to the k and omega equations, respectively. The k augmented term (betaFIK) is added here: https://github.com/mdolab/dafoam/blob/528b2838a751db8ce19337680c9937ccdef6916c/src/adjoint/DAModel/DATurbulenceModel/DAkOmegaSST.C#L775. and the omega augmented term (betaFIOmega) is added here https://github.com/mdolab/dafoam/blob/528b2838a751db8ce19337680c9937ccdef6916c/src/adjoint/DAModel/DATurbulenceModel/DAkOmegaSST.C#L733

**Reply by lpz456** [reply]
Thank you for your patient assistance! I have another question I would like to seek your guidance on. :-)

Based on my understanding, in theory, corrective factors can be applied in arbitrary locations, such as the production term of specific dissipation rate ω (Pω), the destruction term of ω (Dω), or the terms of nonlinear constitutive relationship, etc.

So, if I want to introduce corrective factors like α and β in other locations, such as the terms of nonlinear constitutive relationship, do I need to implement additional programming? Or has similar functionality already been incorporated into the current DAFoam? Determining the derivative of the coefficient matrix of the linear equation system with respect to the corrective factor (dA/dα and dA/dβ) appears to be a key aspect of adjoint optimization.

(I am new to adjoint optimization, and I will read more reference in the near future.)

**Comment by friedenhe** [comment]
If you want to apply the corrected term to other places, the only thing you need to do is to add that term to the corresponding equation. There is no need to do special treatment for the coefficient matrix because DAFoam uses automatic differentiation that can handle the rest for the adjoint.

**Reply by lpz456** [reply]
Thanks for prof's reply.  :)
